Added logic to support showing the Article's Image in the Feed

tfnribeiro opened the pull request:
# New:

## Feed:

### Desktop:
![image](https://github.com/zeeguu/web/assets/17390076/af693641-eb6e-4bfc-a551-17a548aed833)

### Mobile:
![image](https://github.com/zeeguu/web/assets/17390076/cbc8fca9-5d03-4296-b9e5-225b56f4a800)

If an article doesn't have any image links:

### Desktop:
![image](https://github.com/zeeguu/web/assets/17390076/53305b1f-85c5-4f3b-9a6d-7b5a0d37755a)

### Mobile:
![image](https://github.com/zeeguu/web/assets/17390076/81d45fa5-9ded-4452-9343-65b6ff40c7e7)


## Reader
Added preview to the image in the reader:

### Desktop
![image](https://github.com/zeeguu/web/assets/17390076/0b3c4b5a-61d2-4fef-af4a-d0c46e272308)

### Mobile
![image](https://github.com/zeeguu/web/assets/17390076/1a2f57af-c7d6-4e28-ba30-0d8103bf4017)

mircealungu commented on src/articles/ArticlePreview.sc.js:
```diff
@@ -13,6 +13,40 @@ const ArticlePreview = styled.div`
   padding-left: 0.8em;
 `;
+/*
+  The div is encompassed in a
```
encompassed in a ... what? :) now I'm really curious :)

mircealungu commented on src/reader/ArticleReader.js:
```diff
@@ -278,6 +278,20 @@ export default function ArticleReader({ api, teacherArticleID }) {
       </div>
+      <hr></hr>
+      {articleInfo.img_url && (
+        <div style={{ display: "flex", justifyContent: "center" }}>
```
Opportunity for defining a reusable component? As opposed to inline CSS?

tfnribeiro replied:
This is just a center the img div, I am not sure if it extends for other things and I am not sure how useful it would be outside of this case. Do you think it makes sense?

mircealungu replied:
CenteredContent? but you're right - not the most complicated :)

mircealungu commented on src/articles/ArticlePreview.js:
```diff
@@ -105,10 +113,36 @@ export default function ArticleOverview({
+      </s.ArticleContent>
 
+      <div>
+        {!dontShowImage && (
```
now that we have an image, it would make sense to qualify this as being `source_icon` or `source_logo`?

tfnribeiro replied:
I will updated the name to be feed_icon_name

![image](https://github.com/zeeguu/web/assets/17390076/aefaf3f7-c774-4a5a-a649-642fa7e247d7)

mircealungu reviewed (changes requested):
Looks good! Small requests.

tfnribeiro commented:
I have made some changes to make it more clear that the dontShowImage refers to the feed image (now renamed to: `dontShowSourceIcon`)

Removed an unused variable that was not used anymore.

mircealungu reviewed (approved):
looks good. 
i'll merge and deploy.
